napari/_app/_app.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+from functools import lru_cache
+from typing import Dict
+
+from app_model import Application
+
+from ._submenus import SUBMENUS
+from .actions._layer_actions import LAYER_ACTIONS
+from .injection._processors import PROCESSORS
+from .injection._providers import PROVIDERS
+
+
+class NapariApplication(Application):
+    def __init__(self) -> None:
```

Comment:
as in, you specifically want it in the `__init__` instead of calling `self.injection_store.namespace == ...`?  Any reason to prefer one over the other?   I kinda prefer to keep `__init__` methods lean if it's something that has a setter as this one does

Reply:
Ok. I have little different preferences, but they are personal. 

Does the `self.injection_store.register` not depend on provided namespaces?

Reply:
> Does the self.injection_store.register not depend on provided namespaces?

it does, but i'm not sure why that makes a difference?  It's still called first thing

Reply:
you suggest that I could use assignment, that could be called after the constructor is done. But the next line called in the constructor depends on the state of `injections_store`. So after using a new value, it could be broken? Or did I miss something?

Reply:
> you suggest that I could use assignment, that could be called after the constructor is done.

not _after_ the constructor is done... _in_ the constructor.  Basically, the current code looks like this:
```python
class NapariApplication(Application):
    def __init__(self):
        super().__init__(name='napari', raise_synchronous_exceptions=True)
        self.injection_store.namespace = _napari_names
        self.injection_store.register(providers=PROVIDERS, processors=PROCESSORS)
```

I took your comment to mean you'd prefer it to look like this (with a modification of the app-model `__init__` upstream) to set the namespace in _it's_ init

```python
class NapariApplication(Application):
    def __init__(self):
        super().__init__(
            name='napari',
            raise_synchronous_exceptions=True,
            injection_store_namespace=_napari_names
        )
        self.injection_store.register(providers=PROVIDERS, processors=PROCESSORS)
```


those two don't have any runtime difference.  It just requires the addition of a new `__init__` parameter upstream.

Reply:
I think about `__init__` of `NapariApplication`, not `Application`, but maybe again, I overcomplicated it, and writing code ready for all scenarios is not the best idea.

Reply:
but then that just becomes trivially different?
```python
class NapariApplication(Application):
    def __init__(self, namespace=_napari_names):
        super().__init__(name='napari', raise_synchronous_exceptions=True)
        self.injection_store.namespace = namespace
        self.injection_store.register(providers=PROVIDERS, processors=PROCESSORS)
```

this is a solid bike shed :joy: ... they all do the same thing :)